### PR TITLE
GPU only rendering ( OpenGL 4.2 )

### PR DESCRIPTION
**Pull request by bilgili**

### CODE REVIEW

File: livre/eq/render/shaders/fragTexCopy.glsl [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#version 420 core
```
Comment (chevtche) [inline livre/eq/render/shaders/fragTexCopy.glsl]: No copyright?

File: livre/eq/render/RayCastRenderer.cpp [hunk 2]
```diff
@@ -101,18 +103,27 @@ struct RayCastRenderer::Impl
         TransferFunction1D transferFunction;
         initTransferFunction( transferFunction );
 
-        // TODO: Add the shaders from resource directory
-        const int error = _shaders.loadShaders( ShaderData( vertRayCast_glsl,
-                                                            fragRayCast_glsl ));
+        int error = _rayCastShaders.loadShaders( ShaderData( vertRayCast_glsl,
```
Comment (chevtche) [inline livre/eq/render/RayCastRenderer.cpp]: const int error ??

**Comment by eile** [comment]
Does this mean we've abandoned OS X now?

**Comment by chevtche** [comment]
Mac doesn't support OpenGL 4.2 ?

**Comment by bilgili** [comment]
1) If you want to maintain the OSX part I am happy to bring it back. It wont be difficult at all. The API is same, just the impl changed.
2) 4.2 is a 5 years old standard. I am not sure stucking in 4.1 max is a good idea.

**Comment by eile** [comment]
Mac does support OpenGL >3 only using a core profile only. I would guess that Livre still uses non-core functionality somewhere.

We did use Laptops to demo Livre, and others (UZH) use Macs as well. I agree the situation wrt GL on OS X sucks, but I'm not sure if we should abandon the platform.

**Comment by bilgili** [comment]
Lets bring the old renderer back then. Compile this one with the OSX.

**Pull request merged**